src/main/webapp/feed.html

```diff
@@ -23,24 +23,48 @@ <h1>Message Feed</h1>
 
     function MessageFeed() {
       const [allMessages, setMessages] = React.useState([])
+      const [numMessagesPerUser, setNumUserMessages] = React.useState(["No messages!", h('br')])
 
       React.useEffect(fetchMessages, []);
 
       // Fetch messages and add them to the page.
       function fetchMessages() {
+        // store message react components
         const fetchedMessages = []
+        // store how many messages each user posts in map for later use
+        const userMessageCounts = {}
+        // store react components of the num of msgs each user posts
+        const finalUserMessageCounts = ["Number of Messages Per User:", h('br')]
         const url = '/feed';
         fetch(url)
           .then(response => response.json())
           .then(messages => {
+            // get data from each message
             messages.forEach(message => {
+              // if a user has posted before, incr
+              if (message.user in userMessageCounts) userMessageCounts[message.user]++
+              // otherwise set their msg counts to 1
```

Comment:
Please remove these comments. They are just noise since they just repeat what the code is doing.

If you feel the need to explain what code is doing, that is a great opportunity to extract it into a method call:

```js
function getMessageCountsPerUser(messages) { ... };
```

In general, comments should explain *why* code is there, not what the code is doing (because a reader of the code can tell what the code is doing.)